pkg/validator/am0001/default_channel_test.go

```diff
@@ -58,5 +59,24 @@ func TestDefaultChannelInvalid(t *testing.T) {
 				DefaultChannel: "invalid",
 			},
 		},
+		"not present in bundle channels": {
+			AddonMeta: &v1alpha1.AddonMetadataSpec{
+				ID:             "random-operator",
+				DefaultChannel: "alpha",
+				Channels: &[]v1alpha1.Channel{
+					{
+						Name: "alpha",
+					},
+				},
+			},
+			Bundles: []*registry.Bundle{
+				{
+					Annotations: &registry.Annotations{
+						DefaultChannelName: "alpha",
+						Channels:           "beta,stable,rc",
+					},
+				},
+			},
+		},
```

Comment:
You could also add one more test testing the validation if the `DefaultChannel` field doesn't match the `DefaultChannelName` annotation :)